core/corehttp/proxy_test.go

```diff
@@ -0,0 +1,34 @@
+package corehttp
+
+import (
+	"net/http"
+	"strings"
+	"testing"
+
+	"github.com/ipfs/go-ipfs/thirdparty/assert"
+)
+
+func TestParseRequest(t *testing.T) {
+	url := "http://localhost:5001/p2p/QmT8JtU54XSmC38xSb1XHFSMm775VuTeajg7LWWWTAwzxT/http/path/to/index.txt"
+	req, _ := http.NewRequest("GET", url, strings.NewReader(""))
+
+	parsed, err := parseRequest(req)
+	if err != nil {
+		t.Error(err)
+	}
+	assert.True(parsed.httpPath == "path/to/index.txt", t, "proxy request path")
+	assert.True(parsed.name == "/http", t, "proxy request name")
+	assert.True(parsed.target == "QmT8JtU54XSmC38xSb1XHFSMm775VuTeajg7LWWWTAwzxT", t, "proxy request peer-id")
+}
+
+func TestParseRequestInvalidPath(t *testing.T) {
+	url := "http://localhost:5001/p2p/http/foobar"
```

Comment:
Same here, we should test that cases like `/x/http/url` fail as well.

Reply:
/p2p/x/http/url is actually valid here, because it will treat x as the multihash in the short form. It will of course fail later on as soon as the p2p implementation tries to parse it as a multihash.

Reply:
I'm talking about something like ` /p2p/peer/x/http/url`. That is a fail condition in the code that is not tested here (bad format for the long protocol tag mode). I thin you're not covering all the ` return error`  conditions (there are 2 and you're only testing 1 thing).